website/content/api-docs/acl/auth-methods.mdx

```diff
@@ -0,0 +1,392 @@
+  - `BoundAudiences` `(array<string>)` - List of aud claims that are valid for
+    login; any match is sufficient.
+
+  - `AllowedRedirectURIs` `(array<string>)` - A list of allowed values for
+    redirect_uri. Must be non-empty.
+  
+  - `DiscoveryCaPem` `(array<string>)` - PEM encoded CA certs for use by the TLS
+    client used to talk with the OIDC Discovery URL. If not set, system
+    certificates are used.
+
+  - `SigningAlgs` `(array<string>)` - A list of supported signing algorithms.
+    Defaults to `RS256`.
+
+  - `ClaimMappings` `(map[string]string)` - Mappings of claims (key) that will
+    be copied to a metadata field (value). Use this if the claim you are capturing
+    is singular (such as an attribute).
+
+    When mapped, the values in each list can be any of a number, string, or
+    boolean and will all be stringified when returned.
+
+  - `ListClaimMappings` `(map[string]string)` - Mappings of claims (key) will be
+    copied to a metadata field (value). Use this if the claim you are capturing is
+    list-like (such as groups).
+
+### Sample Payload
+
+```json
+{
+  "Name": "example-acl-auth-method",
+  "Type": "OIDC",
+  "TokenLocality": "local",
+  "MaxTokenTTL": "1h0m0s",
+  "Default": false,
+  "Config": {
+    "OIDCDiscoveryURL": "https://my-corp-app-name.auth0.com/",
+    "OIDCClientID": "V1RPi2MYptMV1RPi2MYptMV1RPi2MYpt",
+    "OIDCClientSecret": "example-client-secret",
+    "BoundAudiences": [
+      "V1RPi2MYptMV1RPi2MYptMV1RPi2MYpt"
+    ],
+    "AllowedRedirectURIs": [
+      "http://localhost:4646/oidc/callback"
+    ],
+    "ClaimMappings": {
+      "http://example.com/first_name": "first_name",
+      "http://example.com/last_name": "last_name"
+    },
+    "ListClaimMappings": {
+      "http://consul.com/groups": "groups"
+    }
+  }
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --request POST \
+    --header "X-Nomad-Token: <NOMAD_TOKEN_SECRET_ID>" \
+    --data @payload.json \
+    https://localhost:4646/v1/acl/auth-method
+```
+
+### Sample Response
+
+```json
+{
+    "MaxTokenTTL": "1h0m0s",
+    "Name": "example-acl-auth-method",
+    "Type": "OIDC",
+    "TokenLocality": "local",
+    "Default": false,
+    "Config": {
+        "OIDCDiscoveryURL": "https://my-corp-app-name.auth0.com/",
+        "OIDCClientID": "v1rpi2myptmv1rpi2myptmv1rpi2mypt",
+        "OIDCClientSecret": "example-client-secret",
+        "BoundAudiences": [
+            "v1rpi2myptmv1rpi2myptmv1rpi2mypt"
+        ],
+        "AllowedRedirectURIs": [
+            "http://localhost:4646/oidc/callback"
+        ],
+        "DiscoveryCaPem": null,
+        "SigningAlgs": null,
+        "ClaimMappings": {
+            "http://example.com/first_name": "first_name",
+            "http://example.com/last_name": "last_name"
+        },
+        "ListClaimMappings": {
+            "http://consul.com/groups": "groups"
+        }
+    },
+    "CreateTime": "0001-01-01T00:00:00Z",
+    "ModifyTime": "0001-01-01T00:00:00Z",
```

Comment:
I think it would be nice to have these as non-default time.Time values, to make it feel a little more real.

Reply:
Fixed in 4c7a08eb3737d331051ded3a836a91637128340a